lib/kennel/models/record.rb

```diff
@@ -129,25 +135,53 @@ def build_json
         }.compact
       end
 
+      # One of three outcomes:
+      # - throws (on non-validation error)
+      # - returns, validation_errors is not empty, and as_json is nil (invalid)
+      # - returns, validation_errors is empty, and as_json is not nil (valid)
```

Comment:
seems to repeat the code, maybe replace with a tldr

Reply:
Not sure what you mean. Are you saying that the comment seems superfluous, because the code is easy enough to understand as it is? Or something else?

Reply:
to understand the comment I had to read the code :D